### PR TITLE
"I can smell you." A total rework to Transcendent Olfaction and a new Bloodsucker Olfaction that smells blood

### PR DESCRIPTION
**Pull request by Chubbygummibear**
# Why is this good for the game?
I think current transcendent olfaction is kinda a pain to use being based on fingerprints which has felt like a coinflip whether an item someone used actually has their fingerprint. It's not very interactive because you just get a text message arrow, and gives too much information by letting you sniff a bloody toolbox next to a dead body and says  "You smell John McGrayShit". So now it's more interactive and requires you move around, isn't necessarily a pinpointer because the path might be blocked by areas you don't have access to. Limits your ability to see while following the trail to limit your awareness to the actual world around you. Scents should work more frequently and do give you a good idea of who you're looking for, but does not explicitly give their name away.

# Document the changes in your pull request
Olfaction gets a top to bottom makeover to be more interactive. So here's the briefest summery I can give of it and the bloodsucker version before getting into specifics.

## The short summary
Olfaction is no longer smelling something you're holding for fingerprints, and then getting a "you smell the target north/south/east/west" as long as you have an empty hand

1. You get either Sanguine Olfaction from leveling up as a bloodsucker, or lesser olfaction from genetics

2. Both of these give you a new ability "Acquire a Scent", which looks for scents in a 1 tile circle around you by scanning all objects (including worn and pockets, but not backpacks)

3. If using sanguine olfaction you're sniffing bloodied items, or blood splashes on the floor. If using genetics olfaction, you're looking for "scents" (i'll elaborate on what a "scent" is in a bit, because i killed the fingerprint sniff)

4. After you acquire a scent, you're granted a new ability to "Follow the Scent". For both powers this turns your screen monochrome except for a colored scent trail to follow to your target, as well as highlighting your target in their corresponding color for 20 seconds

5. Staying on the trail as a bloodsucker will give you a speed bonus and consume the trail. Genetics olfaction does not do this, nor does walking on the trail consume it

6. You can recast either of these as desired. Follow the scent will always make a trail to the last target you selected, acquire will always be for selecting a new scent. 

Bloodsucker version:

https://user-images.githubusercontent.com/46236974/203184505-e137b64f-2ec6-444f-b7de-f5ae839effcc.mp4

Genetics version:

https://user-images.githubusercontent.com/46236974/203185807-0c337e7a-dc4a-4fb4-84ae-1db44c832d53.mp4

Genetics version:

## What is a scent?
A stronger but more fragile fingerprint. interacting with items will always leave a scent, but scents are washed away with simple water unlike fingerprints needing to be scrubbed. Additionally, scents do not give away identity immediately, but instead use a scent description provided by your job and species
image
image

## Why does it sniff in a 1-tile circle instead of held item?
The held item system was incredibly clunky because clicking the ability while not realizing you had something in hand would lock you into the prompt and you might break the previous scent you were trying to follow. This also allows users to sniff machinery and structures for people who used it

## How does the scent trail work?
The scent trail uses our a* pathfinding algorithm simplemobs use to find your target. A path is made from you to your target, and your target will leave more scent lines in real time as they move and you have the effect active. HOWEVER, this path is not perfect and will not account for your access when it's made. It may lead you to locked doors, or through areas you shouldn't be in. When the status effect ends and color returns to your screen, you lose the ability to see the scent lines. The color of the scent lines is always crimson for bloodsuckers, but the genetics scent trails will try to use your mutant color first, and if that's too dark or lacking saturation, will try your eye color next. that failing, it will use a random rainbow color.

Genetics:
image

# Spriting
the various scent trails
image

the status effect alert for being speed boosted as a bloodsucker on a blood trail
smells blood icon
dreamseeker_zoKvY7mbQY
the anime reference

i just used the nose for both action icons so one might need to be changed so they're different
image

# Wiki Documentation

I think basically just copy and paste the brief summary at the top for olfaction. I tried to make it pretty comprehensive at a glance.

# Changelog

:cl:  Chubbygummibear, @ToasterBiome, @Skrem7, @LazennG 
rscadd: Transcendent Olfaction Rework 
rscadd: Sanguine Olfaction for bloodsuckers which is Olfaction but for blood
rscdel: Olfaction smelling target north/south/east/west text message gone
tweak: Olfaction picking a scent and following a scent split into two different action buttons instead of one. Additionally picking a scent uses a 1-tile radius circle instead of held item in active hand   
imageadd: New scent trail icon and bloodthirsty status effect icon 
/:cl:

### CODE REVIEW

**Comment by neeshacark** [comment]
cool now its time to bother you for echolocating blind people until you do it

**Comment by Moltijoe** [comment]
I look forward to being singled out as the only preternis

**Comment by ReddicusDragon** [comment]
god this is so cool

**Review by JamieD1 — APPROVED** [review]

File: code/modules/antagonists/bloodsuckers/powers/olfaction.dm [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+		WARNING: it will be difficult to see around you while following a scent trail due to the the color being drained from all your vision except for the trail and your target."
+
+	follow = new /datum/action/bloodsucker/olfaction/follow_scent/lesser()
+
+/datum/action/bloodsucker/olfaction/follow_scent
+	name = "Follow the Scent"
+	desc = "Begin following the scent of your target."
+	button_icon_state = "power_olfac"
+	power_explanation = "<b>Follow the Scent</b>:\n\
+		Activating this Power will create a trail you can follow to the target scent you selected with Olfaction.\n\
```
Comment (ReddicusDragon) [inline code/modules/antagonists/bloodsuckers/powers/olfaction.dm]: Power here should be lowercase, due to grammar stuff. Power isn't a proper noun, that is, person, place or thing.

**Review by ReddicusDragon — CHANGES_REQUESTED** [review]
I've proofread this PR, I have one recommendation

File: code/modules/antagonists/bloodsuckers/powers/olfaction.dm (hunk 1 above)
Comment (Chubbygummibear) [reply code/modules/antagonists/bloodsuckers/powers/olfaction.dm]: got it

**Comment by Chubbygummibear** [comment]
@JamieD1 i'm not sure if you're planning on doing a full merge or test with your approval, but i would recommend test because in addition to being unsure of what new bugs might pop up from more people using it, i did make adjustments to the pathfinding evaluation that *should* only be relevant for players using it, but could possibly have some i n t e r e s t i n g unforeseen interactions with simplemobs pathing

**Comment by JamieD1** [comment]
> @JamieD1 i'm not sure if you're planning on doing a full merge or test with your approval, but i would recommend test because in addition to being unsure of what new bugs might pop up from more people using it, i did make adjustments to the pathfinding evaluation that _should_ only be relevant for players using it, but could possibly have some i n t e r e s t i n g unforeseen interactions with simplemobs pathing

Could we nerf it to 20 seconds, seems a bit long. And sure, can test merge.

**Comment by Chubbygummibear** [comment]
> > @JamieD1 i'm not sure if you're planning on doing a full merge or test with your approval, but i would recommend test because in addition to being unsure of what new bugs might pop up from more people using it, i did make adjustments to the pathfinding evaluation that _should_ only be relevant for players using it, but could possibly have some i n t e r e s t i n g unforeseen interactions with simplemobs pathing
> 
> Could we nerf it to 20 seconds, seems a bit long. And sure, can test merge.

do you mean the duration of the effect? because that is 20 seconds currently

**Comment by JamieD1** [comment]
> > > @JamieD1 i'm not sure if you're planning on doing a full merge or test with your approval, but i would recommend test because in addition to being unsure of what new bugs might pop up from more people using it, i did make adjustments to the pathfinding evaluation that _should_ only be relevant for players using it, but could possibly have some i n t e r e s t i n g unforeseen interactions with simplemobs pathing
> > 
> > 
> > Could we nerf it to 20 seconds, seems a bit long. And sure, can test merge.
> 
> do you mean the duration of the effect? because that is 20 seconds currently

The "follow the scent" has 30 second cooldown

**Comment by Chubbygummibear** [comment]
> > > > @JamieD1 i'm not sure if you're planning on doing a full merge or test with your approval, but i would recommend test because in addition to being unsure of what new bugs might pop up from more people using it, i did make adjustments to the pathfinding evaluation that _should_ only be relevant for players using it, but could possibly have some i n t e r e s t i n g unforeseen interactions with simplemobs pathing
> > > 
> > > 
> > > Could we nerf it to 20 seconds, seems a bit long. And sure, can test merge.
> > 
> > 
> > do you mean the duration of the effect? because that is 20 seconds currently
> 
> The "follow the scent" has 30 second cooldown

follow changed to 20 seconds

**Review by Moltijoe — APPROVED** [review]
I hope you know i plan to change the preternis smell when this gets in

**Comment by ChuChu42** [comment]
as a common transcendent olfaction user, i really like this rework, it was fun to watch you develop this and chat about it in public

why hasnt this been merged yet?

(tough why didnt you make chaplain smell like children like we were talking about)

(at least you made clown smell funny)

**Comment by Chubbygummibear** [comment]
> I hope you know i plan to change the preternis smell when this gets in

lmao that's fine with me. you might have to fight @Skrem7 about it tho :^)

**Comment by Moltijoe** [comment]
> lmao that's fine with me. you might have to fight @Skrem7 about it tho :^)

hold up, did he approve the scent? if so then i guess i'll leave it, just sorta seems like a weird scent for them to have

**Comment by Chubbygummibear** [comment]
> > lmao that's fine with me. you might have to fight @Skrem7 about it tho :^)
> 
> hold up, did he approve the scent? if so then i guess i'll leave it, just sorta seems like a weird scent for them to have

90% of the scent descriptions are his doing, so ye. he did all the species ones and many of the jobs. i only did the really easy/obvious ones

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**